File: flutter_app/lib/data/models/user_model.dart
class UserModel {
  UserModel({
    required this.id,
    required this.username,
    required this.email,
    required this.avatar,
    required this.coins,
    required this.gems,
    required this.wins,
    required this.losses,
    required this.level,
    required this.xp,
  });

  final int id;
  final String username;
  final String email;
  final String avatar;
  final int coins;
  final int gems;
  final int wins;
  final int losses;
  final int level;
  final int xp;

  factory UserModel.fromJson(Map<String, dynamic> json) => UserModel(
        id: json['id'] as int,
        username: json['username'] as String? ?? '',
        email: json['email'] as String? ?? '',
        avatar: json['avatar'] as String? ?? '\u{1F3B2}',
        coins: json['coins'] as int? ?? 0,
        gems: json['gems'] as int? ?? 0,
        wins: json['wins'] as int? ?? 0,
        losses: json['losses'] as int? ?? 0,
        level: json['level'] as int? ?? 1,
        xp: json['xp'] as int? ?? 0,
      );
}

File: backend/src/models/user_model.py
from datetime import datetime

from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Integer, String, UniqueConstraint
from sqlalchemy.orm import relationship

from src.config.database import Base


class User(Base):
    __tablename__ = "ludo_users"

    id = Column(Integer, primary_key=True, index=True)
    username = Column(String(50), unique=True, index=True, nullable=False)
    email = Column(String(120), unique=True, index=True, nullable=False)
    hashed_password = Column(String(255), nullable=False)
    avatar = Column(String(10), nullable=False, default="\U0001F3B2")
    coins = Column(Integer, nullable=False, default=1000)
    gems = Column(Integer, nullable=False, default=50)
    wins = Column(Integer, nullable=False, default=0)
    losses = Column(Integer, nullable=False, default=0)
    level = Column(Integer, nullable=False, default=1)
    xp = Column(Integer, nullable=False, default=0)
    is_active = Column(Boolean, nullable=False, default=True)
    created_at = Column(DateTime(timezone=True), nullable=False, default=datetime.utcnow)

    purchases = relationship("Purchase", back_populates="user", cascade="all, delete-orphan")


class Purchase(Base):
    __tablename__ = "ludo_purchases"
    __table_args__ = (UniqueConstraint("user_id", "item_id", name="uq_user_item"),)

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("ludo_users.id"), nullable=False, index=True)
    item_id = Column(String(50), nullable=False)
    price_coins = Column(Integer, nullable=False, default=0)
    price_gems = Column(Integer, nullable=False, default=0)
    created_at = Column(DateTime(timezone=True), nullable=False, default=datetime.utcnow)

    user = relationship("User", back_populates="purchases")
